docs/ai/quickstarts/build-mcp-server.md

```diff
@@ -0,0 +1,100 @@
+ms.author: alexwolf
+---
+
+# Create and connect to a minimal MCP server using .NET
+
+In this quickstart, you create a minimal Model Context Protocol (MCP) server using the [C# SDK for MCP](https://github.com/modelcontextprotocol/csharp-sdk) and connect to it using GitHub Copilot. MCP servers are services that expose capabilities to clients through the Model Context Protocol (MCP).
+
+## Prerequisites
+
+- [.NET 8.0 SDK or higher](https://dotnet.microsoft.com/download)
+- [Visual Studio Code](https://code.visualstudio.com/)
+- [GitHub Copilot extension](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot) for Visual Studio Code
+
+## Create the project
+
+1. In a terminal window, navigate to the directory where you want to create your app, and create a new console app with the `dotnet new` command:
+
+   ```bash
+   dotnet new console -n MinimalMcpServer
+   ```
+
+1. Navigate to the `MinimalMcpServer` directory:
+
+   ```bash
+   cd MinimalMcpServer
+   ```
+
+1. Add the following NuGet packages to your app:
+
+   ```bash
+   dotnet add package ModelContextProtocol --prerelease
+   dotnet add package Microsoft.Extensions.Hosting
+   ```
+
+    - The [ModelContextProtocol](https://www.nuget.org/packages/ModelContextProtocol) package is the official C# SDK for working with the Model Context Protocol.
+    - The [Microsoft.Extensions.Hosting](https://www.nuget.org/packages/Microsoft.Extensions.Hosting) package provides the generic .NET `HostBuilder` and services for logging and dependency injection.
+
+## Add the app code
+
+Replace the contents of `Program.cs` with the following code to implement a minimal MCP server that exposes simple echo tools. The AI model invokes these tools as necessary to generate responses to user prompts.
+
+:::code language="csharp" source="snippets/mcp-server/program.cs" :::
+
+## Configure the MCP server in Visual Studio Code
+
+Configure GitHub Copilot for Visual Studio Code to use your custom MCP server:
+
+1. If you haven't already, open your project folder in Visual Studio Code.
+1. Create a `.vscode` folder at the root of your project.
+1. Add an `mcp.json` file in the `.vscode` folder with the following content:
+
+   ```json
+   {
+     "inputs": [],
+     "servers": {
+       "MinimalMcpServer": {
+         "type": "stdio",
+         "command": "dotnet",
+         "args": [
+           "run",
+           "--project",
+           "${workspaceFolder}/MinimalMcpServer.csproj"
+         ]
+       }
+     }
+   }
+   ```
+
+1. Save the file.
+
+## Test the MCP server
+
+1. Open GitHub Copilot in Visual Studio Code and switch to agent mode.
+1. Select the **Select tools** icon to verify your **MinimalMcpServer** is available with both tools listed.
+
+    :::image type="content" source="../media/mcp/available-tools.png" alt-text="A screenshot showing the available MCP tools.":::
+
+1. Enter a prompt to run the **ReverseEcho** tool, such as *Reverse the following: "Hello, minimal MCP server!"*
```

Comment:
Since the output is code fenced (line 95), it won't be localized. But this prompt might get localized, so maybe it should be code fenced too?